### DISCUSSION TITLE
Implement Boundary Conditions on the generated Random field

### DISCUSSION THREAD

**Discussion by shivamchb42** [GeoStat-Framework/GSTools#326] [Q&A]
I want to generate a  2D random field and implement boundary conditions at the boundary of the domain. Is there a way included in the library to implement the same? I don't want to just define value at boundaries and generate the field at interior points.

**Comment by LSchueler** [comment]
Hi,

you could try to use conditioned spatial random fields with the conditioning point being the boundary values. Further away from the boundary, the resulting field would still be a truly random field.
[Here](https://geostat-framework.readthedocs.io/projects/gstools/en/latest/examples/06_conditioned_fields/index.html) are examples of how to generate conditioned random fields.
I hope that helps. It would be awesome if you could keep us updated how you get along with this.

Cheers,
Lennart
